PowerValidatedSolutions.psm1

```diff
@@ -51379,46 +51378,134 @@ Function New-AriaNetworksvCenterDataSource {
-		} else {
-				$DataSourceResponse = Invoke-RestMethod -Uri $uri -Method 'POST' -Headers $ariaNetworksHeader -Body $body
-				Write-Output "Successfully added $vcenter to VMware Aria Operations for Networks as a Data Source"
-		}
+        $body = @{
+            fqdn               = $fqdn
+            proxy_id           = $collectorId
+            nickname           = $nickname
+            enabled            = $enabled
+            credentials        = @{
+              username = $username
+              password = $password
+            }
+            ipfix_request      = @{
+              "enable_all"= $true
+            }
+          } | ConvertTo-Json
+        Invoke-RestMethod -Uri $uri -Method 'POST' -Headers $ariaNetworksHeader -Body $body -SkipCertificateCheck
   } Catch {
         Write-Error $_.Exception.Message
   }
 }
 Export-ModuleMember -Function New-AriaNetworksvCenterDataSource
 
+Function New-AriaNetworksNsxtDataSource {
+    <#
+        .SYNOPSIS
+        Add a new NSX-T Data Source to VMware Aria Operations for Networks.
+
+        .DESCRIPTION
+        The New-AriaNetworksNsxtDataSource cmdlet allows a user to add in a NSX-T Manager as a new data source in VMware Aria Operations for Networks in order to collect network flow data.
+
+        .EXAMPLE
+        New-AriaNetworksNsxtDataSource -fqdn sfo-m01-nsx01.sfo.rainpole.io -certificate F:\certs\sfo-m01-nsx01.cer -privatekey F:\certs\sfo-m01-nsx01.key -nickname "sfo-m01-nsx01 - Management Domain NSX Manager" -CollectorId 15832:901:1711011916294613031 -enabled true
+        This example adds a NSX-T Manager server as a new data source in VMware Aria Operations for Networks by using a NSXT-T principal identity user.
+
+        .EXAMPLE
+        New-AriaNetworksNsxtDataSource -fqdn sfo-m01-nsx01.sfo.rainpole.io -username svc-inv-vsphere -password VMw@re1!VMw@re1! -nickname "sfo-m01-nsx01 - Management Domain NSX Manager" -CollectorId 15832:901:1711011916294613031 -enabled true
+        This example adds a NSX-T Manager server as a new data source in VMware Aria Operations for Networks by using a NSX-T service account user with a password.
+
+        .PARAMETER fqdn
+        The NSX-T Manager to add as a data source.
+
+        .PARAMETER certificate
+        The principal identity certificate to use for authentication.
+
+        .PARAMETER privatekey
+        The principal identity private key to use for authentication.
+
+        .PARAMETER username
+        The username to use for authentication.
+
+        .PARAMETER password
+        The password to use for authentication.
+
+        .PARAMETER nickname
+        The nickname to use for this data source in VMware Aria Operations for Networks.
+
+        .PARAMETER collectorId
+        The id of the VMware Aria Operations for Networks collector node where the NSX-T Manager data source will be connected.
+
+        .PARAMETER enabled
+        The parameter to enable the datasource.
+
+    #>
```

Comment:
```suggestion
        The parameter to enable the datasource.
    #>
```

Reply:
Resolved in https://github.com/vmware/power-validated-solutions-for-cloud-foundation/commit/954c74bc51b1fef11dc26b704cc60d52dd1bba80